android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/UploadManager.java

```diff
@@ -169,10 +188,206 @@ private S3Service getS3Service(UploadSession uploadSession) {
         return retrofit.create(S3Service.class);
     }
 
-    public static class UploadFile {
-        public String filename;
-        public String contentType;
-        public long fileLength;
-        public String md5Hash;
+
+    public void addArchive(String filename, Archive archive) {
+
+        LOG.info("Adding archive for " + filename);
+
+        UploadFile uploadFile;
+        try {
+            uploadFile = persist(filename, archive);
+        } catch (Exception e) {
+            e.printStackTrace();
+            return;
+//            throw e;
+        }
+
+        //start async task here
+
+        class ArchiveTask extends AsyncTask<Void, Void, Void> {
```

Comment:
Seems odd to define a class inside a method. Honestly, I didn't even know you could.  I'd suggest moving it out of the method